test/e2e/installplan_e2e_test.go

```diff
@@ -823,6 +845,39 @@ func TestInstallPlanWithCRDSchemaChange(t *testing.T) {
 				return &newCRD
 			}(),
 		},
+		{
+			name:          "existing version is present in new CRD (deprecated field)",
+			expectedPhase: v1alpha1.InstallPlanPhaseComplete,
+			oldCRD: func() *apiextensions.CustomResourceDefinition {
+				oldCRD := newCRD(mainCRDPlural + "d")
+				oldCRD.Spec.Version = "v1alpha1"
+				return &oldCRD
+			}(),
+			newCRD: func() *apiextensions.CustomResourceDefinition {
+				newCRD := newCRD(mainCRDPlural + "d")
+				newCRD.Spec.Version = "v1alpha1"
+				newCRD.Spec.Validation = &apiextensions.CustomResourceValidation{
```

Comment:
Did you try by any chance try and add this validation to newCRD?

Reply:
There is a comparison to determine if the new and old CRDs are different before going ahead of upgrade process. The validation part is here to make sure they are different.